pkg/kubelet/logs/container_log_manager.go

```diff
@@ -349,7 +349,7 @@ func (c *containerLogManager) compressLog(log string) error {
 	defer f.Close()
 	w := gzip.NewWriter(f)
 	defer w.Close()
-	if _, err := io.Copy(w, r); err != nil {
+	if _, err := io.CopyN(w, r, c.policy.MaxSize); err != nil {
```

Comment:
I'm concerned that `CopyN` could cause us to loose information.

In `c.rotateLogs()`, we rotate the log if the files size is > c.policy.maxSize. So for example, if the file is 100KB and the max size is 50KB, we would try and rotate it. However, with this change we would now copy over 50KB of the log.

It seems like this change would only work if we can guarantee that we would rotate the log as soon as it reaches maxSize, which seems extremely difficult given that external processes are writing to the log file.

Reply:
My original motivation for this change is to limit unbounded data flow in kubelet. 
However, I didn't realize that the log file size may be greater c.policy.maxSize.
I will revert this PR. Thanks!